flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/EdgeManagerBuildUtilTest.java

```diff
@@ -0,0 +1,137 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.executiongraph;
+
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.jobgraph.DistributionPattern;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobgraph.tasks.AbstractInvokable;
+import org.apache.flink.runtime.scheduler.SchedulerBase;
+import org.apache.flink.runtime.scheduler.strategy.ConsumerVertexGroup;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.apache.flink.runtime.jobgraph.DistributionPattern.ALL_TO_ALL;
+import static org.apache.flink.runtime.jobgraph.DistributionPattern.POINTWISE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for {@link EdgeManagerBuildUtil} to verify the max number of connecting edges between
+ * vertices for pattern of both {@link DistributionPattern#POINTWISE} and {@link
+ * DistributionPattern#ALL_TO_ALL}.
+ */
+public class EdgeManagerBuildUtilTest {
+
+    @Test
+    public void testPointwise() throws Exception {
+        test(17, 17, POINTWISE);
+        test(17, 23, POINTWISE);
+        test(17, 34, POINTWISE);
+        test(34, 17, POINTWISE);
+        test(23, 17, POINTWISE);
+    }
+
+    @Test
+    public void testAllToAll() throws Exception {
+        test(17, 17, ALL_TO_ALL);
+        test(17, 23, ALL_TO_ALL);
+        test(17, 34, ALL_TO_ALL);
+        test(34, 17, ALL_TO_ALL);
+        test(23, 17, ALL_TO_ALL);
+    }
+
+    private void test(int upstream, int downstream, DistributionPattern pattern) throws Exception {
+
+        Pair<ExecutionJobVertex, ExecutionJobVertex> pair =
+                setupExecutionGraph(upstream, downstream, pattern);
+        ExecutionJobVertex upstreamEJV = pair.getLeft();
+        ExecutionJobVertex downstreamEJV = pair.getRight();
+
+        for (ExecutionVertex ev : upstreamEJV.getTaskVertices()) {
+            assertEquals(1, ev.getProducedPartitions().size());
+
+            IntermediateResultPartition partition =
+                    ev.getProducedPartitions().values().iterator().next();
+            assertEquals(1, partition.getConsumerVertexGroups().size());
+
+            ConsumerVertexGroup consumerVertexGroup = partition.getConsumerVertexGroups().get(0);
+            int actual = consumerVertexGroup.size();
+            int maxNumEdges =
+                    EdgeManagerBuildUtil.getMaxNumEdgesToTarget(upstream, downstream, pattern);
+            switch (pattern) {
```

Comment:
I think we do not need to differentiate ALL_TO_ALL or POINTWISE here.
Instead, we should calculate the actual max edge number and check if it equals to the result of `EdgeManagerBuildUtil.getMaxNumEdgesToTarget`.

Reply:
Yes, that would be better.